### ISSUE TITLE
Allow hiding of any identification

### ISSUE THREAD

**Issue by kueda** [inaturalist/inaturalist#3802]
The taxon you use in an identification can be inappropriate, not just the text that you add to an identification, so we don't think having remarks should be a requirement for hiding an identification, and we don't think hidden identifications should contribute to attributes of the observation, like the community taxon.

- [x] Show the "Hide Content" option for identifications even if there are no remarks
- [x] Remove hidden identifications from consideration when determining the Observation Taxon and the Community Taxon
- [x] Hiding and unhiding an identification should cause the observations to re-calculate the Observation Taxon, Community Taxon, and quality grade (use `moderated_with`, e.g. the way we're [using it](https://github.com/inaturalist/inaturalist/blob/main/app/models/user.rb#L1464) when suspending a user)
- [x] Remove hidden identifications from `/identifications/:login` unless you are viewing your own identifications
- [x] Omit hidden identifications anywhere we count identifications, so user profile, Explore, Activity Item, taxon detail

**Comment by tiwane** [comment]
Just thought about this - should hidden IDs count toward the user's ID count on their profile page? I guess hiding them makes them inactive and thus the IDs wouldn't be counted anyway, correct?

**Comment by kueda** [comment]
Up to you, though we should be careful about omitting hidden idents everywhere if we do that to avoid confusion over discrepancies. Whether or not an identification is hidden is in the index so it should be easy to omit hidden identifications from counts (famous last words).

**Comment by tiwane** [comment]
I think anywhere we count active IDs only we should not count hidden IDs, since they're not active on the observation.

**Comment by kueda** [comment]
This is looking good, Patrick. IMO, the only thing beyond the remaining todo regarding counts is this:

- [ ] The modal preview of the content that will be hidden should show the identification taxon. If there are no remarks, I don't think there's a need to include that empty blockquote.

**Comment by pleary** [comment]
I updated the modal to include a preview of the ID using the ActivityItem component. I also updated it to use the same for comments since it was easy and looked nice - but we can revert that if we don't like it.

As for counts - I've updated the counts I noticed, but I left it unresolved because we show counts in a lot of places and it would be good to get some extra eyes on making sure all relevant counts are updated.
